### DISCUSSION TITLE
Raw SQL driver for queries without the use of JS or Dart Web API for SuperSet support

### DISCUSSION THREAD

**Discussion by Trini8ed** [pocketbase/pocketbase#1799] [Q&A] [answered]
Fairly basic questions here, I'm wanting to connect Apache SuperSet which connects to any SQL database such as MySQL or SQLite to PocketBase.

I'm curious is there a way to expose the SQLlite driver to be accessed without connecting through the web API? I want to be able to use PocketBase with Apache Superset so I have a simple way of visualizing my data.

I could get another service to host the same SQL lite database using the database file (since these would be read-only), but that seems rather redundant to me. This seems like a fairly useful extension of PocketBase if it isn't already supported just add support for sending standard RAW SQL queries directly and just be run with or without the WebAPI. Perhaps just enabled by an extra startup flag.

**Comment by ganigeorgiev** [comment, marked as answer]
I don't understand what do you mean by "raw sql driver".

If you want to connect to the database from external applications, you can open directly the SQLite db file located in `pb_data/data.db` (if I'm understanding correctly, your use case seems to be only for read-only queries and it should be safe to open the same db file from multiple applications/processes).

You also have direct access to the DB layer for executing raw queries when [using PocketBase as framework](https://pocketbase.io/docs/use-as-framework/), if that's what you are asking - https://pocketbase.io/docs/database/.

**Comment by Trini8ed** [comment]
Apologies for the confusion I've never worked with SQLlite before. After doing a bit more investigation I was not aware that there was no host associated with SQLlite it's literally just a db file applications use to communicate to the database vs SQL which listens on a host/port, which I didn't quite understand in my original post. I may have posted this question a bit too preemptively.

I was mainly getting confused by the SQLACHEMY URI from Apache SuperSet making me think otherwise. After looking into what the SQLite URI was I realized it was literally just referring to the file location of the db file. I will go ahead and just provide the DB file directly then. Thank you for the fast response!

![image](https://user-images.githubusercontent.com/1361448/217189085-b9e9d35c-214c-47c6-9c52-3978f2fbfe87.png)
